### ISSUE TITLE
factory.Faker does not properly handle SelfAttribute

### ISSUE THREAD

**Issue by jswinarton** [FactoryBoy/factory_boy#784]
#### Description
According to the documentation, the `factory.SelfAttribute` attribute should be accepted by the `factory.Faker` attribute. However, the example in the documentation doesn't work as expected.

#### To Reproduce
Using the [example in the documentation](https://factoryboy.readthedocs.io/en/latest/reference.html#factory.Faker) with a few modifications is enough to reproduce the error.

Python: 3.7.3
factory-boy: 3.0.1

##### Model / Factory code
```python
import datetime
import factory

class TripFactory(factory.Factory):
    class Meta:
        model = dict

    departure = factory.Faker(
        'date',
        end_datetime=datetime.date.today(),
    )
    arrival = factory.Faker(
        'date_between_dates',
        date_start=factory.SelfAttribute('..departure'),
    )

TripFactory()

```

##### The issue
This produces the following traceback:

```
Traceback (most recent call last):
  File "factorytest.py", line 17, in <module>
    TripFactory()
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/base.py", line 39, in __call__
    return cls.create(**kwargs)
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/base.py", line 527, in create
    return cls._generate(enums.CREATE_STRATEGY, kwargs)
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/base.py", line 464, in _generate
    return step.build()
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/builder.py", line 276, in build
    step.resolve(pre)
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/builder.py", line 217, in resolve
    self.attributes[field_name] = getattr(self.stub, field_name)
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/builder.py", line 382, in __getattr__
    extra=context,
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/faker.py", line 52, in evaluate
    return self.generate(extra)
  File "/tmp/tmpenv/lib/python3.7/site-packages/factory/faker.py", line 49, in generate
    return subfaker.format(self.provider, **kwargs)
  File "/tmp/tmpenv/lib/python3.7/site-packages/faker/generator.py", line 76, in format
    return self.get_formatter(formatter)(*args, **kwargs)
  File "/tmp/tmpenv/lib/python3.7/site-packages/faker/providers/date_time/__init__.py", line 1708, in date_between_dates
    return self.date_time_between_dates(date_start, date_end).date()
  File "/tmp/tmpenv/lib/python3.7/site-packages/faker/providers/date_time/__init__.py", line 1683, in date_time_between_dates
    datetime_to_timestamp(datetime_start),
  File "/tmp/tmpenv/lib/python3.7/site-packages/faker/providers/date_time/__init__.py", line 17, in datetime_to_timestamp
    return timegm(dt.timetuple())
AttributeError: 'SelfAttribute' object has no attribute 'timetuple'
```

It appears that the `SelfAttribute` object is being passed directly to the Faker provider, rather than the value that `SelfAttribute` refers to.

#### Notes
It's also worth noting that the two examples in this section of the documentation have an error. Both inherit from `fatory.Factory` (factory is misspelled). The only things that I changed from the example in the docs are:
  - fix the typo 
  - change the model from `Trip` to `dict`, since I don't have the definition for the `Trip` model.

**Comment by rbarrois** [comment]
Oh, I'm sorry: the doc were showing the version from the current state of the repository, instead of the latest release.

I've just:
- Fixed the configuration on readthedocs;
- Released version 3.1.0 with the updated code.

This should fix the issue; let me know if it doesn't ;)

By the way, thanks a lot for taking the time to fully describe the issue you met, I sincerely appreciate it!

**Comment by jswinarton** [comment]
Looks like it works for us now, thanks for the speedy fix @rbarrois !
